src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs

```diff
@@ -13,22 +13,24 @@ internal abstract partial class AnalyzerDriver : IDisposable
     {
         internal class CompilationData
         {
-            private readonly Dictionary<SyntaxReference, DeclarationAnalysisData> _declarationAnalysisDataMap;
+            private readonly record struct SyntaxReferenceAndSymbol(SyntaxReference Declaration, ISymbol Symbol);
+            private readonly Dictionary<(ISymbol, int), DeclarationAnalysisData> _declarationAnalysisDataMap;
```

Comment:
1. what's the lifetime of hte datamap?  i ask as symbols will root the entire compilation, which is a bit concerning.
2. names for the tuple items (esp. the int).

Reply:
> what's the lifetime of hte datamap?

it’s strongly held by CompilationWithAnalyzers, so matches its lifetime, which is controlled by each analyzer host

> names for the tuple items (esp. the int).

Sure

Reply:
> it’s strongly held by CompilationWithAnalyzers:

Ok great.  can you add a comment to that effect?  thanks!  sign-off stands.